File: java/1475-final-prices-with-a-special-discount-in-a-shop.java
class Solution {
    public int[] finalPrices(int[] prices) {
        int len = prices.length;
        if(len == 0 || len == 1){
            return prices;
        }
        int[] ans = new int[len];
        for(int i=0; i<len; i++){
            ans[i] = prices[i] - findDiscount(prices, len, i+1);
        }
        return ans;
    }
    static int findDiscount(int[] a,int len, int j){
        if(j < len){
            for(int i=j; i<len; i++){
                if( a[i] <= a[j-1] ) {
                    return a[i];
                }
            }
        }
        return 0;
    }
}

File: go/1475-final-prices-with-a-special-discount-in-a-shop.go
func finalPrices(prices []int) []int {
    j := 1
    for i := 0; i < len(prices)-1; i++ {
        if prices[j] <= prices[i] {
            prices[i] = prices[i] - prices[j]
            j = i+2
        } else if j+1 == len(prices) {
            j = i+2
        } else {
            i--
            j++
        }
    }
    return prices
}